### PR TITLE
[Experimental] Minimal MaterialCollectionCell

### PR DESCRIPTION
**Pull request by yarneo**
Related to #2549 

Creating an example that uses the default UICollectionViewController with MaterialCollectionCell. It has the Ink and Shadow capabilities yet doesn't rely on any MaterialCollections (i.e. MDCCollectionView or MDCCollectionViewFlowLayout).

### CODE REVIEW

**Comment by jverkoey** [comment]
Attach screenshots please!

**Comment by yarneo** [comment]
Sure! I just made something super simple that portrays both ink and shadow. 2 Liner cell, with more than one section so the shadow is more obvious. See attached gif.
gif

File: components/CollectionCells/examples/CollectionCellsDefaultUICollectionView.m [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  CGPoint location = [collectionView convertPoint:_inkTouchLocation toView:cell];
+  
+  MDCInkView *inkView;
+  if ([cell respondsToSelector:@selector(inkView)]) {
+    inkView = [cell performSelector:@selector(inkView)];
+  } else {
+    return;
+  }
+
+  [inkView startTouchBeganAnimationAtPoint:location completion:nil];
```
Comment (jverkoey) [inline components/CollectionCells/examples/CollectionCellsDefaultUICollectionView.m:131]: I wonder what a more formal mechanism for associating in with a cell would look like?

**Comment by jverkoey** [comment]
This looks like a really great start. 👍 Nice work!

Based on this exploration, what do you feel like the next steps would be?

**Comment by yarneo** [comment]
My initial thought process was to try and see "how difficult" would it be to use our CollectionCells without relying on our Collections to reach similar results, as it seemed that that could be a concern.

Surprisingly there weren't too many hurdles as you can see. One thing that would be great is if there was a way to possibly improve the ink implementation or have it be configurable / encapsulated in a way that we don't have to do a workaround of displaying it and then un-displaying it in the `didHighlight` `didUnhighlight` methods. The reason we have to do this is if we just set up the ink in initialization, the ink animation is seen also when we scroll (as a tap is recorded there too). Might be a way to see if the gesture is scrolling rather than tapping based on location tracking and be smarter about displaying it or not.

Another thing, is currently we don't have the shadow built into our cells. My concern of adding it is performant mostly and the additional files we would need to add to someone wanting to use our cells but didn't actually want to use any shadow.

An interesting idea would be to subclass the cells and have cells that have different specific behaviors. Such as a cardCell (bigger cards that have shadow), a inkCell, a generic textCell (which we already possess), etc.

That's just my take so far on this initial exploration but of course happy to hear feedback and ideas!

**Comment by ianegordon** [comment]
Something to note is that our standard list items do cast a shadow when in edit mode.  It may make sense to go ahead and add them to our 'standard' cell.

Your goal of encapsulating the ink within the cells to avoid the workaround seems very useful.  If we have that we may be able to provide a basic, re-usable MaterialCell that people could use to implement whatever custom cells they would want.

I'll land this in experimental so @felixaemiliano and others can easily poke around.

**Pull request merged**